[doc] Add DataPipeline + Callbacks + Registry (#207)
PyTorchLightning/lightning-flash: docs/source/general/callback.rst

########
Callback
########

.. _callback:

**************
Flash Callback
**************

> ``FlashCallback`` are an extension of PyTorch Lightning :class:`~pytorch_lightning.callbacks.Callback`.

Reviewer: ```suggestion
:class:`~flash.data.callback.FlashCallback` are extensions of the PyTorch Lightning :class:`~pytorch_lightning.callbacks.Callback`.
```

Reviewer: The lightning link doesn't work, not sure why. Can you maybe find out or replace with a link?

Author: Done !

> 

Reviewer: maybe add more details:
"A callback is a self-contained program that can be reused across projects. You can use any of the following built in callbacks by ...."

Author: Done !

########
Callback
########

.. _callback:

**************
Flash Callback
**************

> :class:`~flash.data.callback.FlashCallback` are extensions of the PyTorch Lightning `Callback <https://pytorch-lightning.readthedocs.io/en/latest/extensions/callbacks.html>`__.

Author: Done !

> :class:`~flash.data.callback.FlashCallback` are extensions of the PyTorch Lightning `Callback <https://pytorch-lightning.readthedocs.io/en/latest/extensions/callbacks.html>`__.

Reviewer 2: ```suggestion
:class:`~flash.data.callback.FlashCallback` is an extension of the PyTorch Lightning `Callback <https://pytorch-lightning.readthedocs.io/en/latest/extensions/callbacks.html>`__.
```

A callback is a self-contained program that can be reused across projects.

> Flash and Lightning have a callback system to execute callbacks when needed.

Reviewer: I would skip this

Author: I am following Lightning docs there.

> Callbacks should capture NON-ESSENTIAL logic that is NOT required for your lightning module to run.

Author: Done !

> 

Reviewer: How are callbacks passed in flash? only using trainer? let's add that

Author: Done !

*******************
Available Callbacks
*******************

BaseDataFetcher
_______________

.. autoclass:: flash.data.callback.BaseDataFetcher
   :members: enable

> BaseViz

Reviewer: I know its long but any reason not to be explicit and call it visualization?

Author: Done !
